js/main.js

```diff
@@ -158,33 +224,42 @@ require(['gitbook', 'jQuery', 'lodash'], function (gitbook, $, _) {
     function filterThreads(section) {
         var words = section.split(' ');
 
-        return _.chain(allThreads)
-            .map(function(thread) {
-                var threadWords = (thread.context.section || '').split(' ');
+        // Compute matching for threads
+        var results = $.map(allThreads, function(thread) {
+            var threadWords = (thread.context.section || '').split(' ');
```

Comment:
@SamyPesse:
To compute matching for threads, we currently use `thread.context.section`, with is the same as `section` passed as an argument to the `filterThreads()` function. Hence, `words` and `threadWords` content is the same.
Didn't you mean to compare the actual `thread.title + thread.body` content to the `section` content?